src/IKVM.Tools.Importer/IkvmImporterInternal.cs

```diff
@@ -1526,6 +1510,35 @@ DiagnosticLevel.Warning when options.errorWarnings.ContainsKey(key) || options.e
             dst.WriteLine(options.path != null ? $"    (in {options.path})" : "");
         }
 
+        internal static void HandleWarnArg(ICollection<string> target, string arg)
+        {
+            foreach (var w in arg.Split(','))
+            {
+                // Strip IKVM prefix
+                int prefixStart = w.StartsWith("IKVM", StringComparison.OrdinalIgnoreCase) ? 4 : 0;
+                int contextIndex = w.IndexOf(':', prefixStart);
+                string context = string.Empty;
+                string parse;
+                if(contextIndex != -1)
+                {
+                    // context includes ':' separator
+                    context = w.Substring(contextIndex);
+                    parse = w.Substring(prefixStart, contextIndex - prefixStart);
+                }
+                else
+                {
+                    parse = w.Substring(prefixStart);
+                }
+
+                if (!int.TryParse(parse, out var intResult))
+                {
+                    // NamedResults aren't supported
```

Comment:
This would've required Reflection, or the .tt-file needs to emit a huge switch for parsing.

Reply:
Yeah, I removed worrying about the name versions, because none of the other tools support it that I could find (csc, etc).

Reply:
SuppressWarnings will probably go away eventually too, but I wasn't exatly sure under what circumstances it was used. It's used, I think, to prevent duplicate warnings/errors from appearing. But it's such a round-about way to go about it.